crates/subspace-networking/src/lib.rs

```diff
@@ -49,6 +49,10 @@ pub use protocols::request_response::handlers::segment_header::{
     SegmentHeaderBySegmentIndexesRequestHandler, SegmentHeaderRequest, SegmentHeaderResponse,
 };
 pub use shared::NewPeerInfo;
+use std::num::NonZeroU16;
 pub use utils::multihash::Multihash;
 pub use utils::prometheus::start_prometheus_metrics_server;
 pub use utils::unique_record_binary_heap::{KeyWrapper, UniqueRecordBinaryHeap};
+
+/// Get piece retry attempts number.
+pub const PIECE_GETTER_RETRY_NUMBER: NonZeroU16 = NonZeroU16::new(3).expect("Not zero; qed");
```

Comment:
If you increase number of retries beyond one you can immediately expect me to ask for rationale and explanation why it is necessary and which issues does it fix.

Also I really don't like non-fundamental values as constants, especially public ones.

Reply:
What do you mean by "non-fundamental values"? 

> If you increase number of retries beyond one you can immediately expect me to ask for rationale and explanation why it is necessary and which issues does it fix.

We discussed that yesterday, here are key takeaways: dsn cache scan without retries has 1-3% "not found" results on my machine. Setting this value to 3 makes scans 100% successful. I set this value to increase the probability of maintaining a healthy cache across the network. I left piece retrieval on block importing and piece reconstruction with their zero retries because they just retry with different piece indexes. In other cases, it makes sense to use the recommended value. Also, if this value changes - we'll change it in a single place.

Reply:
> What do you mean by "non-fundamental values"?

We have fundamental protocol constants like piece size for instance.

Retries though are not fundamental number and should be a part of dynamic configuration. You can set default to some number, but it should still be a configuration parameter, not a constant exposed on `subspace_networking`. As such I don't think it should be a single number in a single place.

So does this mean we're making up to 4 attempts to download a piece (initial + 3 retries)?

Reply:
BTW your last comment is exactly the kind of details I expect to be written in PR description from the beginning. One generic line with zero specifics doesn't cut it.

Reply:
> Retries though are not fundamental number and should be a part of dynamic configuration. You can set default to some number, but it should still be a configuration parameter, not a constant exposed on subspace_networking. As such I don't think it should be a single number in a single place.

Ok. I will change it back to crate's local constants. 

> So does this mean we're making up to 4 attempts to download a piece (initial + 3 retries)?

Yes, in rare cases. In practice, we need at most 2 attempts.